src/rebar_prv_common_test.erl

```diff
@@ -218,9 +218,10 @@ select_tests(_, _, _, {error, _} = Error) -> Error;
 select_tests(State, ProjectApps, CmdOpts, CfgOpts) ->
     %% set application env if sys_config argument is provided
     SysConfigs = sys_config_list(CmdOpts, CfgOpts),
-    Configs = lists:flatmap(fun(Filename) ->
-                                rebar_file_utils:consult_config(State, Filename)
-                            end, SysConfigs),
+    Configs = lists:map(fun(Filename) ->
+                            rebar_file_utils:consult_config(State, Filename)
+                        end, SysConfigs),
+    code:add_pathsa(rebar_state:code_paths(State, all_deps)),
```

Comment:
I'm pretty sure, as I was involved with cover too, that it was about loading modules. No problem adding reset here if you think it'd be necessary.